assemblyline_service_utilities/common/dynamic_service_helper.py

```diff
@@ -2120,6 +2120,25 @@ def get_process_by_pid(self, pid: Optional[int] = None) -> Optional[Process]:
         else:
             return processes_with_pid[0]
 
+    def get_process_by_objectid(self, objectid: Optional[ObjectID] = None) -> Optional[Process]:
+        """
+        This method takes a given ObjectID and returns the associated process
+        :param objectid: The given ObjectID that we want an associated process for
+        :return: The associated process
+        """
+        if not objectid:
+            return None
+
+        process_with_objectid = [
+            process
+            for process in self.get_processes()
+            if process.objectid == objectid
+        ]
```

Comment:
you can avoid processing the whole list here by either using 
``` python
process_with_object_id = next(
    (process for process in self.get_processes() if process.objectid == objectid),
    None)
```
or a for loop with a break statement.